pokete_classes/fightmap.py

```diff
@@ -197,226 +200,234 @@ def get_item(self, items, inv):
-                            continue
-                        players, player = self.choose_poke(figure, players, player, enemy)
-                        attack = ""
-                        break
-                    std_loop(False)
-            else:
-                attack = random.choices(obj.attack_obs,
-                                        weights=[i.ap * ((1.5
-                                                          if enem.type.name in
-                                                                i.type.effective
-                                                          else 0.5
-                                                          if enem.type.name in
-                                                                i.type.ineffective
-                                                          else 1)
-                                                         if info["type"] == "duel"
-                                                         else 1)
-                                                 for i in obj.attack_obs])[0]
+            player = providers[index % 2]
+            enem = providers[(index + 1) % 2]
+
+            attack = player.get_attack(self, enem)
             time.sleep(0.3)
-            if attack != "":
-                obj.attack(attack, enem, self)
+            if attack == "won":
+                return player
+            elif attack != "":
+                player.curr.attack(attack, enem.curr, self, weather)
             self.show()
             time.sleep(0.5)
             winner = None
-            if any(i.hp <= 0 for i in players):
-                winner = [i for i in players if i.hp > 0][0]
-            elif all(i.ap == 0 for i in obj.attack_obs):
-                winner = [i for i in players if i != obj][0]
+            loser = None
+            for prov in providers:
+                if prov.curr.hp <= 0:
+                    winner = providers[(providers.index(prov) + 1) % 2]
+            if all(i.ap == 0 for i in player.curr.attack_obs):
+                winner = providers[(index + 1) % 2]
                 time.sleep(2)
-                self.outp.outp(f"{obj.ext_name} has used all its' attacks!")
+                self.outp.outp(f"{player.curr.ext_name} has used all its' attacks!")
                 time.sleep(3)
             if winner is not None:
-                if (obj.identifier != "__fallback__"
-                        and not winner.player
-                        and any(p.hp > 0 for p in figure.pokes[:6])
-                        and ask_bool(self,
-                                     "Do you want to choose another Pokete?")):
-                    old_player = player
-                    players, player = self.choose_poke(figure, players,
-                                                       player, enemy)
-                    if old_player == player:
+                loser = providers[(providers.index(winner) + 1) % 2]
```

Comment:
You already knew the index in the block above, so why not just save that instead? That will save you having to re-look through the whole list to find the index for the object.